aviatrix/resource_aviatrix_controller_config.go

```diff
@@ -570,21 +606,32 @@ func resourceAviatrixControllerConfigUpdate(d *schema.ResourceData, meta interfa
 		}
 	}
 
-	if d.HasChange("ca_certificate_file_path") || d.HasChange("server_public_certificate_file_path") || d.HasChange("server_private_key_file_path") {
-		certConfig := &goaviatrix.HTTPSCertConfig{
-			CACertificateFilePath:     d.Get("ca_certificate_file_path").(string),
-			ServerCertificateFilePath: d.Get("server_public_certificate_file_path").(string),
-			ServerPrivateKeyFilePath:  d.Get("server_private_key_file_path").(string),
-		}
-		if !(certConfig.CACertificateFilePath != "" && certConfig.ServerCertificateFilePath != "" && certConfig.ServerPrivateKeyFilePath != "") &&
-			!(certConfig.CACertificateFilePath == "" && certConfig.ServerCertificateFilePath == "" && certConfig.ServerPrivateKeyFilePath == "") {
-			return fmt.Errorf("ca_certificate_file_path, server_public_certificate_file_path, and server_private_key_file_path must either all be set or all unset. Please update your configuaration")
-		}
-		err := client.DisableImportedHTTPSCerts()
-		if err != nil {
-			return fmt.Errorf("problem trying to disable imported certs to prepare for importing new certs: %v", err)
-		}
-		if certConfig.CACertificateFilePath != "" && certConfig.ServerCertificateFilePath != "" && certConfig.ServerPrivateKeyFilePath != "" {
+	if d.HasChange("ca_certificate_file_path") || d.HasChange("server_public_certificate_file_path") || d.HasChange("server_private_key_file_path") ||
+		d.HasChange("ca_certificate_file") || d.HasChange("server_public_certificate_file") || d.HasChange("server_private_key_file") {
+		if _, useFilePath := d.GetOk("ca_certificate_file_path"); useFilePath {
+			certConfig := &goaviatrix.HTTPSCertConfig{
+				CACertificateFilePath:     d.Get("ca_certificate_file_path").(string),
+				ServerCertificateFilePath: d.Get("server_public_certificate_file_path").(string),
+				ServerPrivateKeyFilePath:  d.Get("server_private_key_file_path").(string),
+			}
+			err := client.DisableImportedHTTPSCerts()
+			if err != nil {
+				return fmt.Errorf("problem trying to disable imported certs to prepare for importing new certs: %v", err)
```

Comment:
The wording for this error seems a little weird. Maybe something like `could not disable imported certs...` is better.